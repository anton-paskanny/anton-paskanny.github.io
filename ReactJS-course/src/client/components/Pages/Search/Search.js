import React from 'react';

import Header from '../../Header/Header';
import SearchResult from '../../../containers/SearchResult';

const Search = ({ match }) => {
  const { searchVal, searchType } = match.params;
  return (
        <>
            <Header />
            <SearchResult searchPage={true} searchVal={searchVal} searchType={searchType} />
        </>
  );
};

export default Search;
