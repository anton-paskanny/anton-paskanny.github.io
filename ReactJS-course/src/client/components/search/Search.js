import React, { Component } from 'react';

import SearchFilter from '../../containers/SearchFilter';
import SearchInput from './SearchInput/SearchInput';

import { URL_BASE } from '../../utils';

import styles from './styles.css';

class Search extends Component {
    state = {
        searchVal: ''
    }

    handleInputChange = e => {
        this.setState({
            searchVal: e.target.value
        });
    }

    handleSubmit = e => {
        e.preventDefault();

        if (!this.state.searchVal.trim()) {
            return;
        }

        const searchVal = `search=${this.state.searchVal.trim()}&`;
        const searchBy = `searchBy=${this.props.searchType}`;
        const URL = `${URL_BASE}?${searchVal}${searchBy}`;

        this.props.fetchMovies(encodeURI(URL));
    }

    render() {
        return (
            <div className="search-panel">
                <h2 className="search-panel__title">Find your movie</h2>
                <form onSubmit={this.handleSubmit}>
                    <SearchInput
                        value={this.state.searchVal}
                        handleInputChange={this.handleInputChange}
                    />
                    <SearchFilter />
                </form>
            </div>
        )
    }
}

export default Search;

