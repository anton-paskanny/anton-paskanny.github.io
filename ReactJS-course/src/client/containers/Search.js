import { connect } from 'react-redux';
import { fetchMovies } from '../actions/movies';

import Search from '../components/Search/Search';

const mapStateToProps = state => ({
    searchType: state.search.type
});

const mapDispatchToProps = dispatch => ({
    fetchMovies(url) {
        dispatch(fetchMovies(url));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(Search);