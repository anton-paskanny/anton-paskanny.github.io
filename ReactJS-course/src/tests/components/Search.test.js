import React from 'react';
import { shallow, mount } from 'enzyme';

import Search from '../../client/components/Search/Search';

let component, instance, props;

describe('Search', () => {
    beforeEach(() => {
        props = { fetchMovies: jest.fn() };
        component = mount(<Search {...props} />);
        instance = component.instance();
    });

    it('render component correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('should change state on handleInputChange', () => {
        const event = {
            target: {
                value: 'X-men'
            }
        };
        instance.handleInputChange(event);
        expect(component.state().searchVal).toBe(event.target.value);
    });

    it('should change state when search option was changed', () => {
        const event = {
            preventDefault: jest.fn(),
            target: {
                classList: {
                    contains: jest.fn().mockReturnValue(false)
                },
                textContent: 'genres'
            }
        };
        instance.handleSearchByChange(event);
        expect(component.state().searchBy[1].active).toBe(true);
    });

    it('should not change state when active search option was selected', () => {
        const event = {
            preventDefault: jest.fn(),
            target: {
                classList: {
                    contains: jest.fn().mockReturnValue(true)
                },
                textContent: 'genres'
            }
        };
        instance.handleSearchByChange(event);
        expect(component.state().searchBy[0].active).toBe(true);
    });

    it('should fetch new movie on submit event', () => {
        const event = {
            target: {
                value: 'X-men'
            },
            preventDefault: jest.fn()
        };
        instance.handleInputChange(event);
        instance.handleSubmit(event);
        expect(props.fetchMovies).toHaveBeenCalled();
    });

    it('should not fetch new movie on submit event - search value is empty', () => {
        const event = {
            target: {
                value: ''
            },
            preventDefault: jest.fn()
        };
        instance.handleInputChange(event);
        instance.handleSubmit(event);
        expect(props.fetchMovies).not.toHaveBeenCalled();
    });
});
