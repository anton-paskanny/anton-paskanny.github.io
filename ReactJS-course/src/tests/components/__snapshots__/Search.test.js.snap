// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search render component correctly 1`] = `
<Search
  fetchMovies={[MockFunction]}
>
  <div
    className="search-panel"
  >
    <h2
      className="search-panel__title"
    >
      Find your movie
    </h2>
    <form
      onSubmit={[Function]}
    >
      <SearchInput
        handleInputChange={[Function]}
        value=""
      >
        <div
          className="search-panel__input-wrapper"
        >
          <input
            className="search-panel__input"
            name="movie"
            onChange={[Function]}
            value=""
          />
        </div>
      </SearchInput>
      <SearchFilter
        handleSearchByChange={[Function]}
        searchByConfig={
          Array [
            Object {
              "active": true,
              "name": "title",
            },
            Object {
              "active": false,
              "name": "genres",
            },
          ]
        }
      >
        <div
          className="search-filter"
        >
          <div
            className="search-filter__inner-wrapper"
          >
            <p
              className="search-filter__desc"
            >
              Search by
            </p>
            <ul
              className="search-filter__filters"
              onClick={[Function]}
            >
              <li
                className="search-filter__filter-item"
                key="title"
              >
                <button
                  className="search-filter__filter-btn search-filter__filter-btn--active"
                  type="button"
                >
                  title
                </button>
              </li>
              <li
                className="search-filter__filter-item"
                key="genres"
              >
                <button
                  className="search-filter__filter-btn"
                  type="button"
                >
                  genres
                </button>
              </li>
            </ul>
          </div>
          <button
            className="search-filter__search-btn"
            type="submit"
          >
            Search
          </button>
        </div>
      </SearchFilter>
    </form>
  </div>
</Search>
`;
